src/animation/interpolants/SmoothInterpolant.js

```diff
@@ -0,0 +1,103 @@
+/**
+ *
+ * Cubic hermite spline interpolant.
+ *
+ *
+ * @author tschw
+ */
+
+THREE.SmoothInterpolant = function( times, values, stride, result ) {
+
+	THREE.Interpolant.call( this, times, values, stride, result );
+
+};
+
+Object.assign( THREE.SmoothInterpolant.prototype, THREE.Interpolant.prototype, {
+
+	constructor: THREE.SmoothInterpolant,
+
+	DefaultParameters: {
```

Comment:
For another reference here is Unity's AnimationCurve, which is analogous:

http://docs.unity3d.com/ScriptReference/AnimationCurve.html

They have a mode for pre and post, which is more flexible than the loop, pingpong, etc, flag, but I believe it doesn't change the behavior of the curve between the start and end times.